c/frame/repr/text_column.cc

```diff
@@ -183,30 +184,61 @@ static std::string _escaped_char(uint8_t a) {
   return escaped;
 }
 
+static std::string _escape_unicode(int cp) {
+  std::string escaped = (cp <= 0xFF)?   "\\x00" :
+                        (cp <= 0xFFFF)? "\\u0000"
+                                      : "\\U00000000";
+  size_t i = escaped.size() - 1;
+  while (cp) {
+    int digit = cp & 15;
+    escaped[i] = static_cast<char>((digit >= 10)? 'A' + digit - 10
+                                                : '0' + digit);
+    --i;
+    cp >>= 4;
+  }
+  return escaped;
+}
+
 
 std::string Data_TextColumn::_escape_string(const CString& str) {
   ostringstream out;
+  bool allow_unicode = term_->unicode_allowed();
+
   size_t n = static_cast<size_t>(str.size);
-  for (size_t i = 0; i < n; ++i) {
-    auto c = static_cast<unsigned char>(str.ch[i]);
+  auto ch = reinterpret_cast<const unsigned char*>(str.ch);
+  auto end = ch + n;
+  for (;ch < end;) {
```

Comment:
Why not just `while (ch < end)`?

Reply:
evolution :)
You're right though, it should be `while`